File: problemas/iniciante/1153/1153.cs
using System;

class URI {
    static int[] F;

    static int Fatorial(int n){
        if(F[n] == -1){
            F[n] = Fatorial(n - 1) * n;
        }
        return F[n];
    }
    
    static void Main(string[] args) {
        F = new int[14];
        
        F[0] = 1;
        for(int i = 1; i < 14; ++i){
            F[i] = -1;
        }

        int N = int.Parse(Console.ReadLine());

        Console.WriteLine(Fatorial(N));
    }
}

File: problemas/iniciante/1153/1153.jl
F = [-1 for i in 1:12]
F[1] = 1

function fatorial(n)
    if(F[n] == -1)
        global F[n] = n * fatorial(n - 1);
    end
    return F[n]
end

n = parse(Int32, readline())
println(fatorial(n))